### PR TITLE
[python-nextgen] optionally support float strict type

### PR DESCRIPTION
**Pull request by wing328**
- optionally support float strict type via the option "floatStrictType" (default to true)
- add a test

To close https://github.com/OpenAPITools/openapi-generator/issues/14499

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

FYI @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @arun-nalla (2019/11) @spacether (2019/11) @krjakbrjak (2023/02)

### CODE REVIEW

File: bin/configs/python-nextgen-aiohttp.yaml [hunk 1]
```diff
@@ -5,3 +5,4 @@ templateDir: modules/openapi-generator/src/main/resources/python-nextgen
 library: asyncio
 additionalProperties:
   packageName: petstore_api
+  floatStrictType: false
```
Comment (tomplus) [inline bin/configs/python-nextgen-aiohttp.yaml:8]: Is this feature only for asyncio library?

Comment (tomplus) [reply bin/configs/python-nextgen-aiohttp.yaml:8]: OK, I've got it. It's for petstore examples and for aiohttp floatStrict is disabled.

**Review by tomplus — APPROVED** [review]
LGTM

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java [hunk 2]
```diff
@@ -443,12 +449,20 @@ private String getPydanticType(CodegenParameter cp,
                     fieldCustomization.add("multiple_of=" + cp.getMultipleOf());
                 }
 
+                if (floatStrictType) {
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java:452]: This code path is used for json schema number (float + int) and float only processing. Number should allow both in.

How about:
- not adding the strict info when it is number
- adding it when it is float only
- adding tests of both of those cases with and without strict on
- adding a test that verifies the 1.0 ingestion fix

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java:452]: It's an option so users can choose whether they want float in strict type or not.

Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java:452]: Right but that option should only apply to floats right? The way it is now looks like you are applying flat strict when the spec has 'type: number' which should allow in float and int. Did I misunderstand something?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java:452]:
```diff
-    number: confloat(strict=True, le=543.2, ge=32.1) = ...
-   float: Optional[confloat(strict=True, le=987.6, ge=54.3)] = None
-    double: Optional[confloat(strict=True, le=123.4, ge=67.8)] = None
+    number: confloat(le=543.2, ge=32.1) = ...
+    float: Optional[confloat(le=987.6, ge=54.3)] = None
+    double: Optional[confloat(le=123.4, ge=67.8)] = None
```

As illustrated in the sample diff as part of this change, the option (when set to false) simply removes `strict=True`.

Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java:452]: Why are you applying strict float to type number with no format? Won't it throw a validation error if an integer input is given to it? Integer input is valid for that definition.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java:452]: > Why are you applying strict float to type number with no format?

It's a user decision - they can switch on or off strict mode via an option.

You can find more about why some users prefer strict mode in https://docs.pydantic.dev/blog/pydantic-v2/#strict-mode.

**Review by spacether — APPROVED** [review]

**Comment by aimxhaisse** [comment]
Thanks a lot for doing this :)

**Pull request merged**